src/App.tsx

```diff
@@ -1,223 +1,212 @@
+import { Todo } from './types/Todo';
+import { ErrorType } from './types/ErrorType';
+
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorMessage } from './components/ErrorMessage';
+import { TodoForm } from './components/TodoForm';
+
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [sortType, setSortType] = useState<SortType>(SortType.ALL);
+  const [error, setError] = useState('');
+  const [title, setTitle] = useState('');
+  const [selectedId, setSelectedId] = useState<number[]>([]);
+  const [isAdding, setIsAdding] = useState(false);
+
+  const filteredTodo = useMemo(() => todos.filter(todo => {
+    switch (sortType) {
+      case SortType.ALL:
+        return todo;
+      case SortType.ACTIVE:
+        return !todo.completed;
+      case SortType.COMPLETED:
+        return todo.completed;
+      default:
+        return null;
+    }
+  }), [todos, sortType]);
+
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  useEffect(() => {
+    if (!user) {
+      return;
+    }
+
+    const getTodosFromServer = async (userId: number) => {
+      try {
+        const receivedTodos = await getTodos(userId);
+
+        setTodos(receivedTodos);
+      } catch (errorMessage) {
+        setError(`${errorMessage}`);
+      }
+    };
+
+    getTodosFromServer(user.id);
+  }, []);
+
+  const getValue = (element: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(element.target.value);
+  };
+
+  const addNewTodo = useCallback(async (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!title.trim() || !user) {
+      setError(ErrorType.ErrorTitle);
+
+      return;
+    }
+
+    setIsAdding(true);
+
+    try {
+      const newTodo = await createTodo(title, user.id);
+
+      setTodos((prevTodos) => [...prevTodos, newTodo]);
+    } catch {
+      setError(ErrorType.NotAdd);
+    }
+
+    setIsAdding(false);
+    setTitle('');
+  }, [title, user]);
+
+  const removeTodo = useCallback(async (TodoId: number) => {
+    setSelectedId([TodoId]);
+
+    try {
+      await deleteTodos(TodoId);
+
+      setTodos((prevTodos) => prevTodos.filter(({ id }) => id !== TodoId));
+    } catch {
+      setError(ErrorType.NotDelete);
+    }
+  }, [todos, error]);
+
+  const completedTodos = todos.filter(({ completed }) => completed);
+
+  const clearCompletedTodos = useCallback(
+    async () => {
+      try {
+        setSelectedId([...completedTodos].map(({ id }) => id));
+
+        await Promise.all(completedTodos.map(({ id }) => removeTodo(id)));
+        await (() => setTodos((prevTodos) => prevTodos
+          .filter(({ completed }) => !completed)));
```

Comment:
what's the reason of using await here?

Reply:
i have linter error if won't add it

Reply:
let's also call here the function for getting actual todos from server and update them in state instead of doing it manually